include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -0,0 +1,157 @@
+        }
+      }
+      else
+      {
+        // Grab the condition
+        auto const cond     = opts[condition_key];
+
+        // Drop the conditional and process the function call with the universal masker
+        auto const rmv_cond = options{rbr::drop(condition_key, opts)};
+        auto const f        = Func<decltype(rmv_cond)>{rmv_cond};
+
+        // If the conditional call is supported, call it
+        if      constexpr( is_supported ) return detail::mask_op(cond, f, x0, xs...);
+        // if not, call the non-masked version then mask piecewise
+        else if constexpr( any_simd )     return detail::mask_op(cond, detail::eval_2nd, x0, f(x0,xs...));
+      }
+    }
+  };
+
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct constant_callable
+  //!   @brief CRTP base class giving an EVE's @callable the constant function semantic
+  //!
+  //!   **Defined in Header**
+  //!
+  //!   @code
+  //!   #include <eve/module/core.hpp>
+  //!   @endcode
+  //!
+  //!   Constants functions in EVE are built using a very common pattern. Inheriting from eve::constant simplifies the
+  //!   implementation of such eve::callable by just requiring your eve::callable type to implement a static `value`
+  //!   member function that provides the constant value using two parameters:
+  //!     * an eve::options pack containing potential decorators passed to the constant.
+  //!     * an eve::as instance to specify the output type.
+  //!
+  //!   Constant functions in EVE also supports masking, which is directly implemented in eve::constant.
+  //!
+  //!   @note The deferred overload named in the EVE_CALLABLE_OBJECT macro process is still available if an architecture
+  //!   specific implementation of any given constant is required.
+  //!
+  //!   @tparam Tag Type of current @callable being implemented
+  //!
+  //!   @groupheader{Example}
+  //!   @godbolt{doc/traits/callable_constant.cpp}
+  //! @}
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct constant_callable : callable<Func, OptionsValues, conditional_option, Options...>
+  {
+    template<typename O, typename T>
+    constexpr auto behavior(auto arch, O const& opts, as<T> const& target) const
+    {
+      using func_t                =  Func<OptionsValues>;
+      if constexpr( requires{ func_t::deferred_call(arch, opts, target); } )
+      {
+        return func_t::deferred_call(arch, opts, target);
+      }
+      else
+      {
+        // Compute the raw constant
+        auto const constant_value = [&]()
+        {
+          if      constexpr( requires{ Func<OptionsValues>::value(opts, target); } )
```

Comment:
Why is this check? Why not just always one or the other?
Let's do just 2nd and if needed just 1st.